### ISSUE TITLE
Partners Invite - Admin - Edit User

### ISSUE THREAD

**Issue by kathyccheng** [bloom-housing/bloom#1655]
As an admin, 
I might want to edit a user's information 
So that I can make changes if something needs to be changed 

**Acceptance Criteria**
* When I am on the users table, then I will see a link on the user's name 
* When I click on that link, I will be taken to an Edit Screen
* When I change listings access, then the leasing agent will see those changes reflected in their dashboard
* When I change Name or Email, the changes will be saved 
* When I click Resend Invite, then another invite email will be sent 
* When I click Delete User, then I will see an are you sure modal. If I click yes, then the user will be removed. 

**Notes**
* Changing name or email, we will not send any notifications of the change for right now
* Deleting a user is a hard delete

**Designs**
Unconfirmed
https://www.figma.com/file/Y8bFd0aSz7eYJ2TCNid6s3/bloom-partners?node-id=4084%3A169556
Confirmed
https://www.figma.com/file/Y8bFd0aSz7eYJ2TCNid6s3/bloom-partners?node-id=4388%3A175177
Delete
https://www.figma.com/file/Y8bFd0aSz7eYJ2TCNid6s3/bloom-partners?node-id=4388%3A175282

@slowbot add delete and confirm delete to edit screen

### QA Review Notes
1. Go to https://dev-partners-bloom.netlify.app/users logged in as an admin
2. click "Add User"
3. Add the user with the partner role and with an email address you have access to and invite the user
4. check your email and confirm the invite
5. set the users password
6. you should be logged in as that new user and see the listings they were assigned to
7. update the user's assigned listings in another tab logged in as an admin
8. the partner user should see the listing list updated

**Comment by seanmalbert** [comment]
@pbn4 to update with notes for backend updates. We will split off another issue to handle the frontend.

**Comment by pbn4** [comment]
Backend work:

- move user editing his own profile to an endpoint like `PUT /me`
- create a new `PUT /users/:id` for admins only and enable editing any user related property there

**Comment by seanmalbert** [comment]
This is currently blocked by #2045, because the backend is currently using redis to cache listings and the cache is served up regardless if it's public or partners. This prevents a full QA of this feature, because if you edit the listings a user has access to, that does not currently clear the cache. This issue is resolved by 2045, because we will no longer be hitting the cache from partners once that is implemented.

**Comment by kathyccheng** [comment]
Looks good, closing
